### ISSUE TITLE
Load translations from *.po files directly

### ISSUE THREAD

**Issue by Sigvaard** [neris/NGettext#25]
Related issue: https://github.com/neris/NGettext/issues/20

Many programs parse *.po files directly, what's the advantage of *.mo files over *.po files? There's an additional step in maintenance - compilation.

**Comment by neris** [comment]
MO files designed to be easily readable at the runtime. They don't have any problems with encoding, they're easy and really quick to parse.
PO files were designed to be editable by humans. Those are just plain text files, they can be stored in any encoding without any metadata, it's hart to detect it sometimes. The format is pretty complicated to parse and it would be definitely much slower than just reading binary MO file byte-by-byte.

I'm not saying it's impossible. It's pretty much possible, but it's not implemented at the moment since it's not an optimal way and it requires some effort to implement.

Contributions welcome :)

**Comment by Sigvaard** [comment]
@neris but if I'm not wrong MO file is loaded only once and translations are then stored in memory during the application's lifetime so does it make a significant difference?

**Comment by VitaliiTsilnyk** [comment]
You can create a custom MSBuild pre-build task that would call `xgettext`, `msgfmt` and so on, to extract strings and compile `.mo` files. It should be about two lines of code.

I want to implement a loader for `.po` files in NGettext, it's not that hard and it's nice to have, but I can say for sure that I will not have time for that in the nearest future. So contributions are very welcome.

**Comment by ljani** [comment]
> You can create a custom MSBuild pre-build task that would call xgettext, msgfmt and so on, to extract strings and compile .mo files. It should be about two lines of code.

The thing is I don't want to have manual steps to install `xgettext`, etc to be able to compile the solution. All the required functionality should be downloaded as a project dependency. `git clone && dotnet build` should be enough in my opinion.

Sure you could download them, but eh, it gets complex. A proper task would be much cleaner and nicer.

> I want to implement a loader for .po files in NGettext, it's not that hard and it's nice to have, but I can say for sure that I will not have time for that in the nearest future. So contributions are very welcome.

Interesting! Many thanks for everything you've already done with `NGettext`!

**Comment by c4tachan** [comment]
> The thing is I don't want to have manual steps to install `xgettext`, etc to be able to compile the solution. All the required functionality should be downloaded as a project dependency. `git clone && dotnet build` should be enough in my opinion.

It is good practice to include your *.pot and <locale>.po files in your repository. *.mo files should not be included, and should be compiled at build time.

Thankfully, if you configure pre-build steps in your projects, those steps will be included in the repository. This means that `git clone && dotnet build` will automatically execute the pre-build steps. [This stackoverflow question has some answers that will help you](https://stackoverflow.com/questions/40516653/add-prebuild-or-postbuild-to-dotnet-core-projects)
